tests/analysis/adaptation/test_adaptation.py

```diff
@@ -82,3 +87,52 @@ def test_run_benefit_returns_gdf(
             assert _result[f"{_option.id}_benefit"].sum(axis=0) == pytest.approx(
                 _expected[2]
             )
+
+    def test_calculate_bc_ratio_returns_gdf(
+        self,
+    ):
+        # Mocks to avoid complex setup.
+        class MockAdaptation(Adaptation):
+            adaptation_collection: AdaptationOptionCollection = (
+                AdaptationOptionCollection()
+            )
+
+            def __init__(self):
+                pass
+
+        @dataclass
+        class MockAdaptationOption(Adaptation):
+            id: str
+
+        # 1. Define test data.
+        _adaptation = MockAdaptation()
+        _adaptation.adaptation_collection = AdaptationOptionCollection(
+            all_options=[MockAdaptationOption(id=f"Option{x}") for x in range(2)]
+        )
+
```

Comment:
Place this in a fixture, it does not make sense to initialize the `MockAdaptation` with `AdaptationOptionCollection()` to immediately after redefine it as:
```python
AdaptationOptionCollection(
            all_options=[MockAdaptationOption(id=f"Option{x}") for x in range(2)]
        )
```

```suggestion

    @pytest.fixture(name="mocked_adaptation")
    def _get_mocked_adaptation_fixture(self) -> Iterator[Adaptation]:
        class MockAdaptation(Adaptation):
            adaptation_collection: AdaptationOptionCollection = (
                AdaptationOptionCollection(
                    all_options=[MockAdaptationOption(id=f"Option{x}") for x in range(2)]
                ))

            def __init__(self):
                pass

        @dataclass
        class MockAdaptationOption(Adaptation):
            id: str

    def test_calculate_bc_ratio_returns_gdf(
            self,
            mocked_adaptation: Adaptation
        ):
        # 1. Define test data.
        _adaptation = mocked_adaptation

```

Reply:
Nice, works perfect